src/renderer/components/ft-list-video/ft-list-video.js

```diff
@@ -488,6 +516,105 @@ export default Vue.extend({
       })
     },
 
+    avoidChannelBlockerByURI: function () {
+      // Won't hide:
+      //  Subscriptions (#/subscriptions)
+      //  Channel page (#/channel/UCuAXFkgsw1L7xaCfnd5JJOw)
+      //  Your playlists (#/useplaylists)
+      //  History (#/history)
+      const match = window.location.hash.match(/^#\/(subscriptions|channel|userplaylists|history)/)
+      if (match === null) {
+        return false
+      }
+      return true
+    },
+
+    checkIfChannelBlocked: function () {
+      if (!this.useChannelBlocker) {
+        return
+      }
+
+      const channelIndex = this.channelBlockerCache.findIndex((item) => {
+        return item._id === this.channelId
+      })
+
+      if (channelIndex !== -1) {
+        this.channelBlocked = true
+        if (!this.avoidChannelBlockerByURI()) {
+          this.channelBlockerHide()
+        }
+      } else {
+        this.channelBlocked = false
+        this.channelBlockerUnhide()
+      }
+    },
+
+    toggleBlockedChannel: function() {
+      this.channelBlockerIsBlockedById(this.channelId).then(isBlocked => {
+        if (isBlocked) {
+          this.removeChannelFromBlacklist()
+          this.channelBlockerUnhide()
+        } else {
+          this.addChannelToBlacklist()
+          if (!this.avoidChannelBlockerByURI()) {
+            this.channelBlockerHide()
+          }
+        }
+        FtListVideoChannelBlockerEvents.$emit('cbUpdateChannel', this.channelId)
+      }).catch(err => {
+        console.error(err)
+      })
+    },
+
+    addChannelToBlacklist: function () {
+      const channel = {
+        _id: this.channelId,
+        name: this.channelName
+      }
+      this.channelBlockerAddChannel(channel).then(_ => {
+        this.channelBlocked = true
+        this.showToast({
+          message: `"${this.channelName}" ${this.$t('Video.Channel has been added to the block list')}`
+        })
+      }).catch(err => {
+        console.error(err)
+      })
+    },
+
+    removeChannelFromBlacklist: function () {
+      this.channelBlockerRemoveChannelById(this.channelId).then(_ => {
+        this.compactBlockedChannels()
+        this.channelBlocked = false
+        this.showToast({
+          message: `"${this.channelName}" ${this.$t('Video.Channel has been removed from the block list')}`
+        })
+      }).catch(err => {
+        console.error(err)
+      })
+    },
+
+    channelBlockerHide: function () {
+      // Ignore: .watchPlaylistItem
+      // Hide self: .recommendation ("Up Next")
+      // Hide parent: .results (search etc.)
+      const element = this.$el
+      if (element.classList.contains('recommendation')) {
```

Comment:
Again, component hiding should be handing by either filtering the item out of the list before it renders or deleted from the render using `v-if`.